Wise_Owl_Library/Repositories/WiseOwlLibraryDbContext.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Wise_Owl_Library.Models;
+using Microsoft.EntityFrameworkCore;
+using Wise_Owl_Library.Models.Entities;
+using Wise_Owl_Library.Repositories.Interceptors;
+
+namespace Wise_Owl_Library.Repositories
+{
+    public class WiseOwlLibraryDbContext(DbContextOptions<WiseOwlLibraryDbContext> options, PriceChangeInterceptor priceChangeInterceptor) : DbContext(options)
+    {
+        public required DbSet<AuthorEntity> Authors { get; set; }
+        public required DbSet<BookEntity> Books { get; set; }
+        public required DbSet<PriceChangeEntity> PriceChanges { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<BookEntity>()
+                .HasMany(b => b.Authors);
+
+            modelBuilder.Entity<AuthorEntity>()
+                .HasMany(a => a.Books);
+
+            base.OnModelCreating(modelBuilder);
```

Comment:
Mě se ta vazba many to many zda pořád moc na tenhle případ. Zbytečně složité a náchylné na chyby. Při množství záznamu bude efektivnější naimplementovat to bez toho a řešit to jen v repository jednou "neoptimální" query

Reply:
Do produkce bych to asi nedal. Jednodušší kód je lepší kód